src/adapters/fs-sync.ts

```diff
@@ -0,0 +1,56 @@
+import * as fs from 'fs';
+
+import FileSystem from './fs';
+
+import { Entry } from '../types/entries';
+import { Pattern } from '../types/patterns';
+
+export default class FileSystemSync extends FileSystem<Entry[]> {
+	/**
+	 * Use sync API to read entries for Task.
+	 */
+	public read(patterns: string[]): Entry[] {
+		const filepaths = patterns.map(this.getEntryPath, this);
+		const entries = filepaths.map((filepath, index) => this.getEntry(filepath, patterns[index]), this);
+
+		return entries.filter((entry) => entry !== null) as Entry[];
+	}
+
+	/**
+	 * Return entry for the provided filepath.
+	 */
+	public getEntry(filepath: string, pattern: Pattern): Entry | null {
+		try {
+			const stat = this.getStat(filepath);
+
+			return this.makeEntry(stat, pattern);
+		} catch (err) {
+			return null;
+		}
+	}
+
+	/**
+	 * Return stat for the provided filepath.
+	 */
+	public getStat(filepath: string): fs.Stats {
+		const lstat = this.lstat(filepath);
+
+		if (!lstat.isSymbolicLink()) {
+			return lstat;
+		}
+
+		const stat = this.stat(filepath);
```

Comment:
When `stat` throws error we should return result of `lstat` call (maybe this is broken symlink or something).